src/packages/auth/types.ts

```diff
@@ -35,3 +37,16 @@ export type UserResponse = {
 export type User = {
   id: string;
 };
+
+export type ValidateResponse = {
+  success: boolean;
+  message: string;
+  userId?: string;
+  token?: string;
+  redirectURL?: string;
+};
+
+export interface AuthAdapter {
+  validate(req: Request): Promise<ValidateResponse>;
+  logout(req: Request): Promise<boolean>;
```

Comment:
We wanted to avoid forwarding usernames and passwords with these auth adapters, so we instead will redirect to some login UI page that handles user input. I believe Clipper's use case falls under the "CAM API auth + Aerie login UI" case described in the PR description, where the Gateway will redirect users to the Aerie login UI, which just runs the already existing auth flow (i.e. login page hits `/auth/login`, which hits CAM, returns JWT, UI sets cookie, etc)

Reply:
> We wanted to avoid forwarding usernames and passwords with these auth adapters, so we instead will redirect to some login UI page that handles user input.

The UI login page forwards to the Gateway's login endpoint. This endpoint currently has custom logic to have `CAM` auth (L118 in `functions.ts`). It is reasonable to extrapolate from that that Auth Adapters also need to have a `login` method to be used in the `Aerie login UI` case. 

I understand being concerned about forwarding userpasses to the adapters, however I think it will be necessary in order to truly support custom auth adapters.

> I believe Clipper's use case falls under the "CAM API auth + Aerie login UI" case 

I agree, however, I was bringing it up as an example of "CustomAuthAdaptor + Aerie login UI". If they were using a non-CAM auth service, the current implementation of `login` would mean that they would actually have _no_ auth via the UI.

Reply:
We agreed to resolve this as not adding a user-pass based `login` to the interface until we see users making new Auth Handlers want to use it, but to still secure the `login` endpoint so that it doesn't blindly sign users in in the event the current auth type isn't `none` or `cam`.